### PR TITLE
update management policies with GMP changes

### PR DESCRIPTION
**Pull request by lsviben**
Late update for the management policies alpha feature. 

Replaces the old `managementPolicy` to `managementPolicies`, updates the Importing guide and adds a section for `spec.initProvider`. 

Fixes: #497

### CODE REVIEW

**Comment by lsviben** [comment]
Vale seems to complain for `managementPolices`, `deletionPolicy` . Can we add this somewhere for Vale to ignore?

Also btw, I would like to add a guide for using `initProvider` to ignore changes, but as a separate PR.

**Comment by plumbis** [comment]
Re: Vale -
Yes, you can add them to the [crossplane-words.txt](https://github.com/crossplane/docs/blob/master/utils/vale/styles/Crossplane/crossplane-words.txt) file. 
https://docs.crossplane.io/contribute/vale/#spelling-errors-and-exceptions

I have some suggested edits for you but before I make them I want to make sure I understand the feature- 

These changes came in 1.13? 

For `LateInitialize` the resource will set anything defined in `forProvider` and import all other settings? It's like half an observe only resource? 

Are there any restrictions on what can be combined as a policy (even if it seem to make sense)? Is [this chunk from crossplane-runtime](https://github.com/crossplane/crossplane-runtime/blob/0d8cbcef6fac5299f2ce724f3bed0dec2f72f444/pkg/reconciler/managed/policies.go#L46) the supported set of combinations? 

> It's suggested to use a combination of `managementPolicies` without  `LateInitialize` with `initProvider` to avoid late initialization of fields  in `forProvider`.

I'm not following what this is telling me to do/why to do it.

File: content/master/concepts/managed-resources.md
Comment (turkenh) [inline content/master/concepts/managed-resources.md:231]: > importing the external resources not originally created by Crossplane.

Import is not limited to `["Observe"]`. One can still import using `["*"]` as long as they provide all required parameters. Observe only just makes importing more convenient (since you don't have to provide all required params) and safer (since provider doesn't immediately take control).

File: content/master/concepts/managed-resources.md
Comment (turkenh) [inline content/master/concepts/managed-resources.md:233]: Not sure I got what this sentence means 🤔

File: content/master/concepts/managed-resources.md
Comment (turkenh) [inline content/master/concepts/managed-resources.md:243]: A link to a section about Late initialization would be helpful. 
I remember there was such a section under managed resources page but couldn't find it.

File: content/master/concepts/managed-resources.md
Comment (turkenh) [inline content/master/concepts/managed-resources.md:260]: But only when the feature is enabled, right?
It may be hard to follow, which is especially important since it is controlling a critical behavior. Wondering if a table showing the behavior of different combinations would make it clearer.

File: content/master/concepts/managed-resources.md
Comment (turkenh) [inline content/master/concepts/managed-resources.md:264]:
```suggestion
   external resource and update the managed resource spec with late init
```

File: content/master/concepts/managed-resources.md
Comment (turkenh) [inline content/master/concepts/managed-resources.md:274]: What about something like? 

> * `LateInitialize` - Unprovided spec fields are late-initialized to `spec.forProvider` with the default values from the cloud provider. This enables XP to take full control of the external resource, even those values not provided by the user upfront.

Again a link to a more detailed section would be helpful.

**Comment by lsviben** [comment]
> Re: Vale - Yes, you can add them to the [crossplane-words.txt](https://github.com/crossplane/docs/blob/master/utils/vale/styles/Crossplane/crossplane-words.txt) file. https://docs.crossplane.io/contribute/vale/#spelling-errors-and-exceptions
> 

Thanks. Added them.

> I have some suggested edits for you but before I make them I want to make sure I understand the feature-
> 
> These changes came in 1.13?

Yeah, these changes are a part of the Granular managmenent policy / ingore changes that came in 1.13, but are actually implemented on a provider level, so they will be released with the next provider release ( for now just the Upbound official ones).

> 
> For `LateInitialize` the resource will set anything defined in `forProvider` and import all other settings? It's like half an observe only resource?

Late initialization has been around for a while, its not a part of this feature. But its related because this feature now allows us to skip Late Initialization by setting up a granular management policy which does not include it. A big part of the ignore changes feature requests came because Crossplane was late-initializing some fields that users wanted to be left alone. 

I saw that Late Initialize section was removed from the docs, probably unintentionally, so I found an old version from the repo and returned it back. Maybe it could do some rework, to avoid confusion on what it actually does. 

> 
> Are there any restrictions on what can be combined as a policy (even if it seem to make sense)? Is [this chunk from crossplane-runtime](https://github.com/crossplane/crossplane-runtime/blob/0d8cbcef6fac5299f2ce724f3bed0dec2f72f444/pkg/reconciler/managed/policies.go#L46) the supported set of combinations?

yeah we support by default a certain set of policies, which is subject to change if somebody has a good use case. The providers have an option to set their own set of polices to support if they prefer so.

> 
> > It's suggested to use a combination of `managementPolicies` without  `LateInitialize` with `initProvider` to avoid late initialization of fields  in `forProvider`.
> 
> I'm not following what this is telling me to do/why to do it.

Reworded, basically it means that in most cases you should use a mangement policy which does not have `LateInitialize` when using `initProviders` because the whole point of `initProviders` is to avoid setting the fields in `forProvider`. Which can be messed up by Crossplane late-initializing that field in `forProvider`

File: content/master/concepts/managed-resources.md
Comment (lsviben) [reply content/master/concepts/managed-resources.md]: added the table, and mention that its only when the feature is enabled.

File: content/master/concepts/managed-resources.md
Comment (lsviben) [reply content/master/concepts/managed-resources.md]: there was a section in the old docs, I found it and brought it back as it was. Maybe it could do with some refurbishment?

File: content/master/concepts/managed-resources.md
Comment (plumbis) [inline content/master/concepts/managed-resources.md:325]: Suggested restructure to improve wording and flow and add `{{<hover>}}` support.

```suggestion
{{<hint "important" >}}
The managed resource `initProvider` option is an alpha feature related to 
[managementPolicies]({{<ref "./managed-resources#managementpolicies" >}}).

Enable the `initProvider` in a provider with `--enable-management-policies`
in a
[ControllerConfig]({{<ref "./providers#controller-configuration" >}}).
{{< /hint >}}

The `spec.initProvider` is a subset of the `spec.forProvider` fields that
Crossplane merges with `forProvider` to create the external resource, but
ignores when updating it. This allows users to create a resource with all the
required fields which may then be controlled externally. For example, a user may
want to create a `NodeGroup` with the required `desiredSize` field, but then
let an autoscaler control the field. If the `desiredSize` field is included
in the`forProvider` field, Crossplane will attempt to update the field, which
will conflict with the autoscaler.
```

```
The 
{{<hover label="initProvider" line="7">}}initProvider{{</hover>}} defines 
settings Crossplane applies only when creating a new managed resource.  
Crossplane ignores settings defined in the 
{{<hover label="initProvider" line="7">}}initProvider{{</hover>}} 
field that change after creation.

{{<hint "note" >}}
Settings in `forProvider` are always enforced by Crossplane. Crossplane reverts
any changes to a `forProvider` field.  

Settings in `initProvider` aren't enforced by Crossplane. Crossplane ignores any
changes to a `initProvider` field.
{{</hint >}}

Using `initProvider` is useful for setting initial values that a Provider may
automatically change, like an auto scaling group.

For example, creating a 
{{<hover label="initProvider" line="2">}}NodeGroup{{</hover>}} 
with an initial 
{{<hover label="initProvider" line="9">}}desiredSize{{</hover>}}.  
Crossplane doesn't change the 
{{<hover label="initProvider" line="9">}}desiredSize{{</hover>}}
setting back when the Provider auto scales the Node Group.

{{< hint "tip" >}}
Crossplane recommends configuring
{{<hover label="initProvider" line="6">}}managementPolicies{{</hover>}} without
`LateInitialize` to avoid conflicts with `initProvider` settings.
{{< /hint >}}
```

File: content/master/concepts/managed-resources.md
Comment (plumbis) [inline content/master/concepts/managed-resources.md:346]: I moved this to a `{{<hint "tip">}}` above the code block.

**Review by plumbis — CHANGES_REQUESTED** [review]
Thanks for all the work Lovro!  There are a lot of suggestions but it's mainly big chunks of moving things around. 

Let me know if you have any questions or objections to my suggestions.

File: content/master/concepts/managed-resources.md
Comment (lsviben) [reply content/master/concepts/managed-resources.md]: Applied, just added a slight change to

```
Settings in `forProvider` are always enforced by Crossplane. Crossplane reverts
any changes to a `forProvider` field.  

Settings in `initProvider` aren't enforced by Crossplane. Crossplane ignores any
changes to a `initProvider` field.
```
to 
```
Settings in `forProvider` are always enforced by Crossplane. Crossplane reverts
any changes to a `forProvider` field in the external resource.

Settings in `initProvider` aren't enforced by Crossplane. Crossplane ignores any
changes to a `initProvider` field in the external resource.
```

To make a clear distinction in making changes in managed its external resource

File: content/master/concepts/managed-resources.md
Comment (lsviben) [reply content/master/concepts/managed-resources.md]: btw this table in dark mode is unreadable, do you know how to fix that?

Comment (plumbis) [reply content/master/concepts/managed-resources.md]: Well that's a CSS design bug 😆

Can you add this:
```
    //Striped table font color
    .table-striped-columns > :not(caption) > tr > :nth-child(even){
        color: inherit !important;
    }
```

before the final closing brace `}`

to  `themes/geekboot/assets/scss/dark-mode.scss`

Comment (lsviben) [reply content/master/concepts/managed-resources.md]: thanks, looks good now

**Review by plumbis — APPROVED** [review]
changes to fix vale and improve table display. Otherwise LGTM. Approving, assuming there's no problems with the suggestions.

File: content/master/concepts/managed-resources.md
Comment (turkenh) [reply content/master/concepts/managed-resources.md]: Pushed a commit to fix that.

**Review by turkenh — APPROVED** [review]
Thanks @lsviben, looking good to me!

I just pushed a couple of comments to fix Vale errors and make some improvements in the late initialization section.

@plumbis feel free to have another look, or please let me know if we can hit the merge button.

**Pull request merged**
